server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -299,6 +302,8 @@ Runnable getGlobalCheckpointSyncer() {
     private final RefreshPendingLocationListener refreshPendingLocationListener;
     private volatile boolean useRetentionLeasesInPeerRecovery;
 
+    private final CheckpointRefreshListener checkpointRefreshListener;
+
```

Comment:
Lets just use the abstraction here `ReferenceManager.RefreshListener` instead of a concrete  `CheckpointRefreshListener ` that would allow us to inject the `IndexShard` with a `NoopCheckpointRefreshListener` if segrep is disabled